File: src/main/java/RemoveElement/RemoveElement.py
# same logic as MoveZeroElement
# use index i to scan all elements
# use pos to count the length of elements not equal to target val
class Solution:
    def removeElement(self, nums: List[int], val: int) -> int:
        pos=0
        for i in range(len(nums)):
            if nums[i] != val:
                nums[pos]=nums[i]
                pos+=1
        return pos

File: src/main/java/RemoveElement/LonlyBonze/RemoveElement.java
class RemoveElement {
	public int removeElement(int[] nums, int val) {	       
		int i=0,j=nums.length-1,count=0;
		while(i<=j){
			if(nums[i]==val){
				while(j>i&&nums[j]==val){
					j--;
					count++;
				}
				if(j==i){
					count++;
					break;
				}else{
					nums[i]=nums[j];
					j--;
					count++;
				}
			}
			i++;
		}
		return nums.length-count;
	}
}
